from PyQt5.uic import loadUi
from PyQt5.QtWidgets import QDialog


class UpdateView(QDialog):
    def __init__(self):
        super(UpdateView, self).__init__()
        loadUi('./resources/ui/dialog/update.ui', self)

        # Connect things
        self.buttonBox.accepted.connect(self.buttonBoxAccepted)

        self.labelTitle.setText('Update RExploit')

        self.setResult("")
        self.buttonBox.setEnabled(False)

        self.setProgressBarState(0)

    def buttonBoxAccepted(self):
        self.accept()

    def setProgressBarState(self, state):
        if state == 0:
            self.progressBar.hide()
            self.progressBar.setRange(0, 1)
            self.progressBar.setValue(0)
        elif state == 1:
            self.progressBar.setFormat("Waiting")
            self.progressBar.setRange(0, 1)
            self.progressBar.setValue(0)
        elif state == 2:
            self.progressBar.setFormat("Updating")
            self.progressBar.setRange(0, 0)
            self.progressBar.show()
        elif state == 3:
            self.progressBar.setFormat("Completed")
            self.progressBar.setRange(0, 1)
            self.progressBar.setValue(1)
            self.buttonBox.setEnabled(True)
            self.progressBar.show()
        elif state == 4:
            self.progressBar.setFormat("Error")
            self.progressBar.setRange(0, 1)
            self.progressBar.setValue(0)
            self.progressBar.show()

    def setResult(self, result):
        self.labelResult.setText(result)
